validator/iam_allow_ban_roles.rego

```diff
@@ -0,0 +1,59 @@
+#
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+package templates.gcp.GCPIAMAllowBanRolesConstraintV1
+
+import data.validator.gcp.lib as lib
+
+deny[{
+	"msg": message,
+	"details": metadata,
+}] {
+	constraint := input.constraint
+	lib.get_constraint_params(constraint, params)
+	asset := input.asset
+
+	binding := asset.iam_policy.bindings[_]
+	role := binding.role
+
+	matches_found = {r | r := role; glob.match(params.roles[_], [], r)}
+
+	mode := lib.get_default(params, "mode", "allow")
+
+	target_match_count(mode, desired_count)
+	count(matches_found) != desired_count
+
+	message := sprintf("IAM policy for %v grants %v", [asset.name, role])
```

Comment:
Add note about whether the rule was a deny or allow

Reply:
Error message updated. The message now clearly states whether the violation is due to role being banned or not allowed.